pkg/northbound/server.go

```diff
@@ -118,6 +130,23 @@ func (s *Server) Serve(started func(string)) error {
 		return err
 	}
 	opts := []grpc.ServerOption{grpc.Creds(credentials.NewTLS(tlsCfg))}
+	if s.cfg.SecurityCfg.AuthenticationEnabled {
```

Comment:
if `s.cfg.SecurityCfg` is nil, will this panic?

Reply:
If we pass an empty struct, it won't be considered as nil since it will initialized by default values (here with false) and doesn't panic. we also cannot also pass nil to the northbound.NewServerConfig function because  it does not accept type nil (i.e. the type is SecurityConfig) so at least we should pass an empty struct. I don't know if there is any other cases.